DOC-122/Document EnterpriseGraphs (#1054)
arangodb/docs: 3.10/graphs-enterprise-graphs-getting-started.md

that you want to migrate over to be an EnterpriseGraph to benefit from
the sharding strategy. In our example the graph has only two collections:
`old_vertices` which is a document collection and `old_edges` which is the
corresponding edge collection.

**Export general-graph data**
Now we need to export the raw data of those
collections using `arangoexport`:

    arangoexport --type jsonl --collection  old_vertices --output-directory docOutput --overwrite true
    arangoexport --type jsonl --collection  old_edges --output-directory docOutput --overwrite true

Note here: We are using the `jsonl` type as it is most flexible and can be
used with larger datasets. The `json` type is limited in amount of documents,
as it cannot be parsed line by line. The `csv` and `tsv` formats are also fine,
but require to define the list of attributes to export. `jsonl` exports data
as is, and due to it's line based layout can be processed line by line and
therefore has no artifical restrictions on the data.
After this step we there are two files in `docOutput` folder, that should look
like this:

1) `docOutput/old_vertices.jsonl`:

>     {"_key":"Alice","_id":"old_vertices/Alice","_rev":"_edwXFGm---","attribute1":"value1"}

Reviewer: Text in a list that wraps around should be indented like the first character after the bullet point, and fenced code blocks should be indented like this too, including the triple backticks. Adjusted by https://github.com/arangodb/docs/pull/1054/commits/2a57778de76f0471dcb267f0b169dc0b6b95900d

On the edges EnterpriseGraphs disallow to store the `_key` value, so we need to remove this attribute
on import:

    arangoimport --collection old_edges --file docOutput/old_edges.jsonl --remove-attribute "_key"

And that's it we are done, the graph has been migrated.

**Import data while changing collection names**

In this variant we changed the collection names.
Let us asume we have renamed `old_vertices` to `vertices`
For the vertex data this change is not relevant, the `_id` values will adjust themselfes,
so we can again just import the data, and just target the new collection name:

    arangoimport --collection vertices --file docOutput/old_vertices.jsonl

For the edges we need to apply more changes, as we need to rewire them.
First thing, we again have to remove the `_key` value as it is disallowed for enterprise graphs.
Second, we have changed the name of the `_from` collection, so we need to provide a `from-collection-prefix`.
Third, the same for the `_to` collection. Note here: We can only change all vertices on `_from` resp. `_to`
side with this mechanism. However we can use different collections on `_from` and `_to`.

> Fourth, in order to make the change of vertex collection actually happy we need to allow "overwrite-collection-prefix"

Reviewer: ```suggestion
Fourth, in order to make the change of vertex collection actually happy we need to allow `--overwrite-collection-prefix`
```

> if this flag is not set, only values without any given collection will be changed, this is helpful if your data is not

Reviewer: ```suggestion
if this option is not set, only values without any given collection are changed. This is helpful if your data is not
```
